operator/src/main/java/org/keycloak/operator/controllers/KeycloakService.java

```diff
@@ -41,15 +41,22 @@ public KeycloakService(KubernetesClient client, Keycloak keycloakCR) {
     }
 
     private ServiceSpec getServiceSpec() {
-        String name = isTlsConfigured(keycloak) ? Constants.KEYCLOAK_HTTPS_PORT_NAME : Constants.KEYCLOAK_HTTP_PORT_NAME;
-        return new ServiceSpecBuilder()
-              .addNewPort()
-              .withPort(getServicePort(keycloak))
-              .withName(name)
-              .withProtocol(Constants.KEYCLOAK_SERVICE_PROTOCOL)
-              .endPort()
-              .withSelector(getInstanceLabels())
-              .build();
+        var builder = new ServiceSpecBuilder().withSelector(getInstanceLabels());
+
+        boolean tlsConfigured = isTlsConfigured(keycloak);
+        Optional<HttpSpec> httpSpec = Optional.ofNullable(keycloak.getSpec().getHttpSpec());
+        boolean httpEnabled = httpSpec.map(HttpSpec::getHttpEnabled).orElse(false);
+        if (!tlsConfigured || httpEnabled) {
+            builder.addNewPort()
+                    .withPort(httpSpec.map(HttpSpec::getHttpPort).orElse(Constants.KEYCLOAK_HTTP_PORT))
+                    .withName(Constants.KEYCLOAK_HTTP_PORT_NAME).withProtocol(Constants.KEYCLOAK_SERVICE_PROTOCOL)
+                    .endPort();
+        }
+        if (tlsConfigured) {
+            builder.addNewPort().withPort(getServicePort(keycloak)).withName(Constants.KEYCLOAK_HTTPS_PORT_NAME)
```

Comment:
I think we don't need to use the `getServicePort(keycloak)` which checks again if TLS is configured.

Reply:
We don't, it's just to prevent more code duplication.  I'll refactor getServicePort a little more.